logstash-core/src/main/java/org/logstash/common/io/DeadLetterQueueWriter.java

```diff
@@ -220,6 +255,88 @@ private void innerWriteEntry(DLQEntry entry) throws IOException {
         lastWrite = Instant.now();
     }
 
+    private void executeAgeRetentionPolicy() throws IOException {
+        if (isOldestSegmentExpired()) {
+            deleteExpiredSegments();
+        }
+    }
+
+    private boolean isOldestSegmentExpired() {
+        if (retentionTime == null) {
+            return false;
+        }
+        final Instant now = clock.instant();
+        return oldestSegmentTimestamp
+                .map(t -> t.toInstant().isBefore(now.minus(retentionTime)))
+                .orElse(false);
+    }
+
+    private void deleteExpiredSegments() throws IOException {
+        // remove all the old segments that verifies the age retention condition
+        boolean cleanNextSegment;
+        do {
+            if (oldestSegmentPath.isPresent()) {
+                Path beheadedSegment = oldestSegmentPath.get();
+                deleteTailSegment(beheadedSegment);
+            }
+            updateOldestSegmentReference();
+            cleanNextSegment = isOldestSegmentExpired();
+        } while (cleanNextSegment);
+    }
+
+    private void deleteTailSegment(Path segment) throws IOException {
+        final long segmentSize = Files.size(segment);
+        this.currentQueueSize.add(-segmentSize);
+        try {
+            Files.delete(segment);
+            logger.debug("Deleted exceeded retained age segment file {}", segment);
+        } catch (NoSuchFileException nsfex) {
+            // the last segment was deleted by another process, maybe the reader that's cleaning consumed segments
+            logger.info("File not found {}, maybe removed by the reader pipeline", segment);
```

Comment:
This maybe a pretty common occurrence if `clean_consumed` is being used. 

Do we know that this pipeline is able to be "cleaned" by the reader, and if so, can we make this debug

Reply:
With PR #14256 when the reader is used with `clean_consumed` feature then it's created also a lock file. So we could rely on this side effect of the reader to understand if there is a reader that's clean consumed segments, but 
- we can't rely on it, because maybe the writer is started before the reader pipeline, and so if we do the check at writer creation we could potentially deduct a false condition.
- we use side effect to deduct reader's behavior, which create a subtle dependency between writer and reader that's not immediately visible. So future works on this stuff could be harder than expected.

Reply:
However I think we could move the `info` to `debug` also for non clean_consumed use-case.

Reply:
Fixed by 837884dccfc1371167d914a94589a2f821601ea2